nearpy/tests/engine_tests.py

```diff
@@ -73,5 +73,68 @@ def test_retrieval_sparse(self):
             self.assertEqual(y_data, x_data)
             self.assertAlmostEqual(y_distance, 0.0, delta=delta)
 
+    def test_delete_vector(self):
+        # test case for 1 hash
+        dim = 5
+        name_hash = 'testHash'
+        unibucket = UniBucket(name_hash)
+        engine = Engine(dim, lshashes=[unibucket])
+        # Index 20 random vectors (set their data to a unique string)
+        for index in range(20):
+            v = numpy.random.randn(dim)
+            engine.store_vector(v, index)
+        # Create random query vector
+        keys = set()
+        bucket = engine.storage.buckets[name_hash][name_hash]
+        keys = {i[1] for i in bucket}
+
+        self.assertEqual(len(keys), 20)
+
+        engine.delete_vector(15) #delete the vector with key = 15
+
+        new_keys = set()
+        bucket = engine.storage.buckets[name_hash][name_hash]
+        new_keys = {i[1] for i in bucket}
+
+        self.assertTrue(len(keys) > len(new_keys))
+        self.assertEqual(len(new_keys), 19) #new keys has 19 elements instead of 20
+
+        self.assertTrue(15 in keys)
+        self.assertFalse(15 in new_keys) # the key 15 is the one missing
+
+    def test_delete_vector2(self):
+        # test case for several hashes
+        dim = 5
+        hashes = []
+        for k in xrange(10): # 10 hashes
```

Comment:
Aw, sorry for that. I just made a bunch of pushes... :sob:  Ok, I will wait for some minutes. thanks for your comments anyway! :+1: